apis/compute/v1alpha2/types.go

```diff
@@ -34,12 +40,53 @@ const (
 	DefaultNodeCount = 1
 )
 
+// Error strings
+const (
+	errResourceIsNotAKSCluster = "The managed resource is not a AKSCluster"
+)
+
+// ResourceGroupNameReferencerForAKSCluster is an attribute referencer that resolves name from a referenced ResourceGroup
+type ResourceGroupNameReferencerForAKSCluster struct {
+	apisv1alpha2.ResourceGroupNameReferencer `json:",inline"`
+}
+
+// Assign assigns the retrieved group name to the managed resource
+func (v *ResourceGroupNameReferencerForAKSCluster) Assign(res resource.CanReference, value string) error {
+	aks, ok := res.(*AKSCluster)
+	if !ok {
+		return errors.Errorf(errResourceIsNotAKSCluster)
+	}
+
+	aks.Spec.ResourceGroupName = value
+	return nil
+}
+
+// SubnetIDReferencerForAKSCluster is an attribute referencer that resolves name from a referenced ResourceGroup
+type SubnetIDReferencerForAKSCluster struct {
+	networkv1alpha2.SubnetIDReferencer `json:",inline"`
+}
+
+// Assign assigns the retrieved group name to the managed resource
+func (v *SubnetIDReferencerForAKSCluster) Assign(res resource.CanReference, value string) error {
+	aks, ok := res.(*AKSCluster)
+	if !ok {
+		return errors.Errorf(errResourceIsNotAKSCluster)
+	}
+
+	aks.Spec.VnetSubnetID = value
+	return nil
+}
+
 // AKSClusterParameters define the desired state of an Azure Kubernetes Engine
 // cluster.
 type AKSClusterParameters struct {
 	// ResourceGroupName is the name of the resource group that the cluster will
 	// be created in
-	ResourceGroupName string `json:"resourceGroupName"`
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
```

Comment:
Should this have `// +optional` as well? It looks like it is on all `omitempty` fields except for the reference fields. I don't know exactly how the mechanics of the `// +optional` tag works, so I may be off-base here.

Reply:
I avoided adding `+optional` comment markers to types that don't already use them - we should probably add them wholesale during a v1beta1 port.